update/update.go

```diff
@@ -24,6 +24,11 @@ type StateEntry struct {
 
 // CheckForUpdate checks whether this software has had a newer release on GitHub
 func CheckForUpdate(client *api.Client, stateFilePath, repo, currentVersion string) (*ReleaseInfo, error) {
+	stateEntry, err := getStateEntry(stateFilePath)
+	if err == nil && time.Since(stateEntry.CheckedForUpdateAt).Hours() < 24 {
+		return nil, nil
+	}
```

Comment:
Wouldn't this be simpler to just change line 47 to return `nil, nil`? Unless I have missed something it seems to be the same behavior.

Reply:
I've pushed an update addressing this. Bear in mind that it'll still have to retrieve `StateEntry` from file twice every time it performs a version check over HTTP since it now writes the latest update to the file instead of returning it.